public/js/components/Expressions.js

```diff
@@ -3,66 +3,111 @@ const { connect } = require("react-redux");
 const { bindActionCreators } = require("redux");
 const ImPropTypes = require("react-immutable-proptypes");
 // const classnames = require("classnames");
+const Svg = require("./utils/Svg");
 const actions = require("../actions");
 const { getExpressions, getPause } = require("../selectors");
-// const ObjectInspector = React.createFactory(require("./ObjectInspector"));
+const Rep = React.createFactory(require("./Rep"));
 // const { truncateStr } = require("../utils/utils");
 const { DOM: dom, PropTypes } = React;
 
 require("./Expressions.css");
 
 const Expressions = React.createClass({
   propTypes: {
-    pauseInfo: ImPropTypes.map,
     expressions: ImPropTypes.list,
     addExpression: PropTypes.func,
     updateExpression: PropTypes.func,
-    evaluateExpression: PropTypes.func,
-    loadObjectProperties: PropTypes.func,
-    command: PropTypes.func
+    deleteExpression: PropTypes.func
   },
 
   displayName: "Expressions",
 
   addExpression(e) {
+    const { addExpression } = this.props;
     if (e.key === "Enter") {
       const expression = {
         input: e.target.value
       };
       if (e.target.id) {
         expression.id = e.target.id.split("-").pop();
       }
-      this.props.addExpression(expression);
+      e.target.value = "";
+      addExpression(expression);
     }
   },
 
   updateExpression(e) {
+    const { updateExpression } = this.props;
+    const id = e.target.id.split("-").pop();
     const expression = {
-      id: e.target.id.split("-").pop(),
-      input: e.target.textContent.split(" --> ")[0]
+      id,
+      input: e.target.textContent
     };
-    this.props.updateExpression(expression);
+    updateExpression(expression);
+    setTimeout(() => {
+      document.getElementById("expressionInputId-" + id).focus();
+    }, 1);
+  },
+
+  renderExpressionValue(value) {
+    if (!value) {
+      return;
+    }
+    if (value.exception) {
+      return;
+    }
+    if (typeof value.result !== "object") {
+      return value.result;
+    }
+    if (typeof value.result === "object") {
+      return Rep({ object: value.result });
+    }
+  },
+
+  deleteExpression(expression, e) {
+    e.stopPropagation();
+    const { deleteExpression } = this.props;
+    deleteExpression(expression);
   },
 
   renderExpression(expression) {
     return dom.div(
       { className: "expression-container",
-        key: expression.id },
+      key: expression.id },
       expression.updating ?
-        dom.input(
-          { type: "text",
-            className: "input-expression",
-            id: "expressionInput-" + expression.id,
-            placeholder: "Add watch Expression",
-            onKeyPress: this.addExpression,
-            defaultValue: expression.input }
+        dom.span(
```

Comment:
Nit, but this could be: `renderExpressionContainer`, `renderExpressionUpdating`, `renderExpression`